core/templates/services/svg-sanitizer.service.ts

```diff
@@ -187,15 +212,6 @@ export class SvgSanitizerService {
       return false;
     }
 
-    // Check for malicious SVG.
-    const { tags: invalidTags, attrs: invalidAttributes } = (
-      this.getInvalidSvgTagsAndAttrsFromDataUri(base64ImageData));
-
-    if (invalidTags.length > 0 || invalidAttributes.length > 0) {
-      return false;
-    }
-
-    // The SVG is safe and valid.
     return true;
```

Comment:
@IamEzio note that the function name (and comment above it) suggests that it will return true only when the base64SVG is _valid_ and _safe_. 

I suspect this logic was removed to fix the issue that was causing invalid images to appear broken. Based on looking at how "isValidBase64Svg" is used:

https://github.com/oppia/oppia/blob/46b3f446dda612d1434a00c4529546333bfaf8e6/core/templates/services/svg-sanitizer.service.ts#L214-L218

If so, I think what can be done here is to change the function to "isBase64Svg" and remove parts in the comments that refer to validation of tags/attrs. Then add the sanitization logic inside `getTrustedSvgResourceUrl` before invoking `this.sanitizer.bypassSecurityTrustResourceUrl`. WDYT @IamEzio?

Reply:
> @IamEzio note that the function name (and comment above it) suggests that it will return true only when the base64SVG is _valid_ and _safe_.
> 
> I suspect this logic was removed to fix the issue that was causing invalid images to appear broken. Based on looking at how "isValidBase64Svg" is used:
> 
> https://github.com/oppia/oppia/blob/46b3f446dda612d1434a00c4529546333bfaf8e6/core/templates/services/svg-sanitizer.service.ts#L214-L218
> 
> If so, I think what can be done here is to change the function to "isBase64Svg" and remove parts in the comments that refer to validation of tags/attrs. Then add the sanitization logic inside `getTrustedSvgResourceUrl` before invoking `this.sanitizer.bypassSecurityTrustResourceUrl`. WDYT @IamEzio?

This actually might be a better way and also avoid calling of `removeTagsAndAttributes` explicitly. However, we would then have to update `invalidTagsAndAttributes` property before calling `getTrustedSvgResourceUrl` (i,e. On image upload rather than on Image save in some case) to get information about which tags and attributes we have removed.

This does seem to be a better approach. Will implement this and inform you of the other changes.